Remotion/Web/UnitTests/Core/WebStringTest.cs

```diff
@@ -0,0 +1,254 @@
+    }
+
+    [Test]
+    public void Write_UnencodedWebString_EncodesOutputWithTransformedLineBreaks ()
+    {
+      var stringValue = "aoe   \" & ' < > é \r \n \r\n";
+      var webString = WebString.CreateFromText (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (webString.Write);
+      Assert.That (renderedString, Is.EqualTo ("aoe &#160; &quot; &amp; &#39; &lt; &gt; &#233; <br /> <br /> <br />"));
+    }
+
+    [Test]
+    public void Write_NullEncodedWebString_OutputsEmptyString ()
+    {
+      var webString = WebString.CreateFromHtml (null);
+
+      var renderedString = ExecuteWithHtmlTextWriter (webString.Write);
+      Assert.That (renderedString, Is.EqualTo (string.Empty));
+    }
+
+    [Test]
+    public void Write_EmptyEncodedWebString_OutputsEmptyString ()
+    {
+      var webString = WebString.CreateFromHtml (string.Empty);
+
+      var renderedString = ExecuteWithHtmlTextWriter (webString.Write);
+      Assert.That (renderedString, Is.EqualTo (string.Empty));
+    }
+
+    [Test]
+    public void Write_EncodedWebString_DoesNotEncodeOutput ()
+    {
+      var stringValue = "aoe   \" & ' < > é \r \n \r\n";
+      var webString = WebString.CreateFromHtml (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (webString.Write);
+      Assert.That (renderedString, Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void AddAttribute_NullUnencodedWebString_RendersEmptyAttribute ()
+    {
+      var webString = WebString.CreateFromText (null);
+
+      var renderedString = ExecuteWithHtmlTextWriter (
+          writer =>
+          {
+            webString.AddAttribute (writer, HtmlTextWriterAttribute.Class);
+            writer.RenderBeginTag ("a");
+          });
+      Assert.That (renderedString, Is.EqualTo ("<a class=\"\">"));
+    }
+
+    [Test]
+    public void AddAttribute_EmptyUnencodedWebString_RendersEmptyAttribute ()
+    {
+      var webString = WebString.CreateFromText (string.Empty);
+
+      var renderedString = ExecuteWithHtmlTextWriter (
+          writer =>
+          {
+            webString.AddAttribute (writer, HtmlTextWriterAttribute.Class);
+            writer.RenderBeginTag ("a");
+          });
+      Assert.That (renderedString, Is.EqualTo ("<a class=\"\">"));
+    }
+
+    [Test]
+    public void AddAttribute_UnencodedWebStringWithAttributeEnum_EncodesAttributeValue ()
+    {
+      var stringValue = "aoe   \" & ' < > é \r \n \r\n";
+      var webString = WebString.CreateFromText (stringValue);
+
+      var renderedString = ExecuteWithHtmlTextWriter (
+          writer =>
+          {
+            webString.AddAttribute (writer, HtmlTextWriterAttribute.Class);
+            writer.RenderBeginTag ("a");
+          });
+      Assert.That (renderedString, Is.EqualTo ("<a class=\"aoe   &quot; &amp; &#39; &lt; > é \r \n \r\n\">"));
+    }
+
+    [Test]
+    public void AddAttribute_UnencodedWebStringWithAttributeEnumThatHasEncodeDefault_EncodeOutput ()
```

Comment:
Tests in line 137 and 152 appear to have identical code? Am I missing something?